chips/nrf52832/src/interrupt_service.rs

```diff
@@ -1,21 +1,30 @@
-use crate::gpio;
-use nrf52::interrupt_service::InterruptService;
+use crate::deferred_call_tasks::DeferredCallTask;
+use nrf52::chip::Nrf52DefaultPeripherals;
 
-pub struct Nrf52832InterruptService {
-    nrf52: nrf52::interrupt_service::Nrf52InterruptService<'static>,
+/// This struct, when initialized, instantiates all peripheral drivers for the nrf52840.
+/// If a board wishes to use only a subset of these peripherals, this
+/// should not be used or imported, and a modified version should be
+/// constructed manually in main.rs.
+pub struct Nrf52832DefaultPeripherals<'a> {
+    pub nrf52: Nrf52DefaultPeripherals<'a>,
+    // put additional 52832 specific peripherals here
 }
-
-impl Nrf52832InterruptService {
-    pub unsafe fn new() -> Nrf52832InterruptService {
-        Nrf52832InterruptService {
-            nrf52: nrf52::interrupt_service::Nrf52InterruptService::new(&gpio::PORT),
+impl<'a> Nrf52832DefaultPeripherals<'a> {
+    pub fn new(ppi: &'a crate::ppi::Ppi) -> Self {
+        Self {
+            nrf52: unsafe { Nrf52DefaultPeripherals::new(&crate::gpio::PORT, ppi) },
```

Comment:
So, why is `Nrf52DefaultPeripherals::new` unsafe but not `Nrf52832DefaultPeripherals::new`? This `unsafe` block deserves an explanation why it's safe.

Reply:
Hmm..`Nrf52DefaultPeripherals::new()` is unsafe because it instantiates the MPU, syscalls, and systicks, all functions which are marked unsafe. I should probably just bubble that unsafe all the way to main for now, though eventually it might be better for us to rethink whether those constructors are actually unsafe in the rust sense.

Changed `Nrf52832DefaultPeripherals::new()` and `Nrf52840DefaultPeripherals::new()` to be marked unsafe.

Reply:
Whether the underlying function should be `unsafe` is a question we can indeed ask. But I don't see any reason to mark the relatively simple `Nrf52832DefaultPeripherals::new` wrapper function as safe (unless the unsafety comes fom the GPIO port, and that using `&crate::gpio::PORT` suddenly makes it safe).

> I should probably just bubble that unsafe all the way to main for now.

I think that's how `unsafe` should work, unless there is a clear reason why some intermediate function is always safe to call, by guaranteeing that the safety invariants are always respected. As far as I understand it, these invariants are currently only guaranteed in the main function.